Add Music Track Indicator plugin

ShaunDreclin opened the pull request:
Shows chat notifications when unlocking music tracks

Resolves #3697

![screenshot](https://user-images.githubusercontent.com/5851433/53607353-57711600-3b8c-11e9-87bc-e25f6d47fa86.gif)

ShaunDreclin commented on runelite-client/src/main/java/net/runelite/client/plugins/musicindicator/MusicIndicatorPlugin.java:
```diff
@@ -0,0 +1,159 @@
+
+		if (musicTrackIDs.isEmpty())
+		{
+			for (int id : client.getEnum(819).getKeys())
```
Hmm probably a good idea. `MUSIC_TRACK_NAMES` and `MUSIC_TRACK_IDS`?

Adam- commented:
Definitely add comments on most of this since the entire music system seems to involve a lot of layers of indirection. `MusicTrackVarp` seems to me like it could just be replaced with a list of varps? I will see about adding an index into var changed event so we can maybe do only one `switch` to check if something interesting has changed, since I think this event fires alot.

ShaunDreclin commented:
Added comments describing the maps and modified the rest to hopefully be more self explanatory, changed the enum to a list, and used the new `.getIndex()` method to tidy up the `VarbitChanged` event. Wasn't able to use a switch statement but this is still quite a bit tidier
